Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -536,13 +588,90 @@ protected override void ProcessRecord()
             InstancesResource.ImportRequest request = Service.Instances.Import(body, Project, Instance);
             Operation result = request.Execute();
             result = WaitForSqlOperation(result);
-            if (result.Error != null) {
+            if (UploadLocalFile)
+            {
+                bucketService.Objects.Delete(bucketName, fileName).Execute();
+                bucketService.Buckets.Delete(bucketName).Execute();
+            }
+            if (result.Error != null)
+            {
                 foreach (OperationError error in result.Error.Errors)
                 {
+
                     throw new GoogleApiException("Google Cloud SQL API", error.Message + error.Code);
                 }
             }
         }
+
+        private string UploadFile()
+        {
+            Random rnd = new Random();
+            int bucketRnd = rnd.Next(1000000);
+            bucketName = "import" + bucketRnd.ToString();
+            fileName = "toImport";
+            bucketService = new StorageService(GetBaseClientServiceInitializer());
+            CreateBucket();
+            try
+            {
+                UploadObject();
+            }
+            catch (Exception e)
+            {
+                bucketService.Buckets.Delete(bucketName).Execute();
+                throw e;
+            }
+            UpdatePermissions();
```

Comment:
Setting one of the default ACLs doesn't quite work. The instance has is own service email that must be used in order for the import operation to not fail. General project privacy does not give instances in your project access.